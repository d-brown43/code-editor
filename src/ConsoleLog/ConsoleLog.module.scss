$line-height: 1.4em;

.container {
  display: flex;
  flex-direction: column;
  padding-left: 40px;
  padding-right: 40px;
}

.line {
  min-height: $line-height;
}

.line:not(:last-of-type) {
  border-bottom: 1px solid #eee;
}

.formattedMessage {
  margin: 0;
}

.wrapper {
  height: 10 * $line-height;
  overflow-y: auto;
}