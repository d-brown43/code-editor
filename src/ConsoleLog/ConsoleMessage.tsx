import * as React from "react";
import styles from "./ConsoleLog.module.scss";


type ConsoleMessage = {
    message: any;
}

const ConsoleMessage = ({message}: ConsoleMessage) => {
    const formattedMessage = React.useMemo(() => (
        <pre className={styles.formattedMessage}>
            {JSON.stringify(message, null, 2)}
        </pre>
    ), [message]);
    if (typeof message === 'string') {
        return <span>{message}</span>;
    }
    return formattedMessage;
};

export default ConsoleMessage;