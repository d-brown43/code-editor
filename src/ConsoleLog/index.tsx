import * as React from "react";
import styles from './ConsoleLog.module.scss';
import ConsoleMessage from "./ConsoleMessage";
import ReactResizeDetector from 'react-resize-detector';

type ConsoleLog = {
    logMessages: any[];
};

const ConsoleLog = ({logMessages}: ConsoleLog) => {
    const [height, setHeight] = React.useState(0);
    const wrapper = React.useRef<HTMLDivElement>(null);

    React.useEffect(() => {
        if (wrapper.current !== null) {
            wrapper.current.scrollTop = height - wrapper.current.offsetHeight;
        }
    }, [height]);

    return (
        <div
            className={styles.wrapper}
            ref={wrapper}
        >
            <div className={styles.container}>
                <ReactResizeDetector
                    handleHeight
                    onResize={(width, height) => {
                        setHeight(height);
                    }}
                />
                {logMessages.map((message, index) => (
                    <div
                        key={index}
                        className={styles.line}
                    >
                        <ConsoleMessage
                            message={message}
                        />
                    </div>
                ))}
            </div>
        </div>
    );
};

export default ConsoleLog;