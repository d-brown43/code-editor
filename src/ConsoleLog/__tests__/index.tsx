import ConsoleLog from "../index";
import {shallow} from "enzyme";
import * as React from "react";
import {render, cleanup} from "@testing-library/react";

it('matches snapshots', () => {
    const consoleLog = shallow(
        <ConsoleLog
            logMessages={[{
                message: 'Hello World',
                type: "log"
            }]}
        />
    );
    expect(consoleLog).toMatchSnapshot();
});

describe('behaviour', () => {
    afterEach(cleanup);

    it('renders given content', () => {
        const {getByText} = render(
            <ConsoleLog
                logMessages={[
                    {
                        message: 'Hello World',
                        type: "log",
                    },
                    {
                        message: 'Hello 2',
                        type: "log",
                    }
                ]}
            />
        );
        expect(() => getByText(/Hello World/)).not.toThrow();
        expect(() => getByText(/Hello 2/)).not.toThrow();
    });
});
