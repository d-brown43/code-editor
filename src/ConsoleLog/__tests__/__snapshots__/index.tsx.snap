// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshots 1`] = `
<div
  className="wrapper"
>
  <div
    className="container"
  >
    <div
      className="line"
      key="0"
    >
      <ConsoleMessage
        message="Hello World"
      />
    </div>
    <div
      className="line"
      key="2"
    >
      <ConsoleMessage
        message=""
      />
    </div>
    <div
      className="line"
      key="3"
    >
      <ConsoleMessage
        message=""
      />
    </div>
    <div
      className="line"
      key="4"
    >
      <ConsoleMessage
        message=""
      />
    </div>
  </div>
</div>
`;
